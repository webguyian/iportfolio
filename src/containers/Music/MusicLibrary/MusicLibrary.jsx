import React from 'react';

import Link from 'components/Link/Link';
import Loading from 'components/Loading/Loading';
import Text from 'components/Text/Text';

import { ALBUM_URL } from 'modules/music/constants';
import { useMusic } from 'modules/music/hooks';

const MusicLibrary = () => {
  const tracks = useMusic();
  const baseClass = 'music-library';
  const trackClass = `${baseClass}-track`;

  return (
    <div className={baseClass}>
      <Text className={`${baseClass}-title`} element="h1" type="display">
        Library
      </Text>
      <Text className={`${baseClass}-subtitle`} element="h2" type="display">
        Recently Added
      </Text>
      {tracks.length ? (
        <ul className={`${baseClass}-tracks`}>
          {tracks.map(track => (
            <li key={track.id}>
              <div className={trackClass}>
                <Link to="/music/player" state={{ track }}>
                  <img
                    className={`${trackClass}-cover`}
                    alt={`${track.albumName} album cover`}
                    src={`${ALBUM_URL}/${track.albumId}/images/160x160.jpg`}
                  />
                </Link>
                <Text className={`${trackClass}-name`} element="h3">
                  {track.name}
                </Text>
                <Text className={`${trackClass}-artist`} element="h4">
                  {track.artistName}
                </Text>
              </div>
            </li>
          ))}
        </ul>
      ) : (
        <Loading label="Loading tracks..." />
      )}
    </div>
  );
};

export default MusicLibrary;
