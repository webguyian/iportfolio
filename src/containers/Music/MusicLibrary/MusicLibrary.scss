.music-library {
  @include stretch-column-nowrap;
  padding: 0 $gutter;
  overflow-y: auto;

  &-title,
  &-subtitle {
    margin-bottom: $gutter;
  }

  &-subtitle {
    color: $pink;
    font-size: $font-size-l;
  }

  &-tracks {
    @include unstyled-list;
    display: flex;
    flex-flow: row wrap;
    justify-content: space-between;
  }

  &-track {
    max-width: 160px;
    min-height: 200px;
    margin-bottom: $gutter-m;
    background-repeat: no-repeat;

    &-cover {
      display: block;
      width: 150px;
      height: 150px;
      border-radius: $border-radius-m;
      background-color: $gray-dark;
      text-align: center;
    }

    &-name,
    &-artist {
      margin: $gutter-s 0 0;
      font-size: $font-size-s;
    }

    &-artist {
      color: $gray;
    }
  }
}
