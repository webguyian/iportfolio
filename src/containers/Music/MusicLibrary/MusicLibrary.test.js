import React from 'react';
import { act, render } from '@testing-library/react';

import MusicLibrary from './MusicLibrary';

describe('<MusicLibrary />', () => {
  it('renders correctly', async () => {
    let component;

    await act(async () => {
      const result = render(<MusicLibrary />);

      component = result.asFragment();
    });

    expect(component).toMatchSnapshot();
  });
});
