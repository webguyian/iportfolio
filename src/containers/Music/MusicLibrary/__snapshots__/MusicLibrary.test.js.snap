// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MusicLibrary /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="music-library"
  >
    <h1
      class="ui-text ui-text--display music-library-title"
    >
      Library
    </h1>
    <h2
      class="ui-text ui-text--display music-library-subtitle"
    >
      Recently Added
    </h2>
    <div
      class="loading-wrapper"
    >
      <span
        class="ui-text ui-text--body"
      >
        Loading tracks...
      </span>
      <div
        class="loading"
      >
        <div
          class="loading-tick"
        />
        <div
          class="loading-tick"
        />
        <div
          class="loading-tick"
        />
        <div
          class="loading-tick"
        />
        <div
          class="loading-tick"
        />
        <div
          class="loading-tick"
        />
        <div
          class="loading-tick"
        />
        <div
          class="loading-tick"
        />
        <div
          class="loading-tick"
        />
        <div
          class="loading-tick"
        />
        <div
          class="loading-tick"
        />
        <div
          class="loading-tick"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
